builtin/game/chat.lua

```diff
@@ -982,24 +982,66 @@ core.register_chatcommand("days", {
 	end
 })
 
+local function parse_shutdown_param(param)
+	local delay, reconnect, message
+	local one, two, three
+	one, two, three = param:match("^(%S+) +(%-r) +(.*)")
+	if one and two and three then
+		-- delay, reconnect and message
+		delay, reconnect, message = one, two, three
+	else
+		-- two arguments
+		one, two = param:match("^(%S+) +(.*)")
+		local done = false
+		if one and two then
+			if tonumber(one) then
+				delay = one
+				if two == "-r" then
+					reconnect = two
+				else
+					message = two
+				end
+				done = true
+			elseif one == "-r" then
+				reconnect, message = one, two
+				done = true
+			end
+		end
+		-- one argument
+		if not done then
+			one = param:match("(.*)")
+			if tonumber(one) then
+				delay = one
+			elseif one == "-r" then
+				reconnect = one
+			else
+				message = one
+			end
+		end
+	end
+	return delay, reconnect, message
```

Comment:
You moved it but the logic is just as hard to follow as before.
By "early return" I mean replacing line 991 with `return one, two, three`, the rest of function can then be taken out of the else clause (same thing in other applicable places)